File: piwigo/Image/Manage/Cells/EditImageTagsTableViewCell.m
//
//  EditImageTagsTableViewCell.m
//  piwigo
//
//  Created by Spencer Baker on 2/18/15.
//  Copyright (c) 2015 bakercrew. All rights reserved.
//

#import "EditImageTagsTableViewCell.h"
#import "TagsData.h"

@interface EditImageTagsTableViewCell()

@property (weak, nonatomic) IBOutlet UILabel *tagsLabel;
@property (weak, nonatomic) IBOutlet UILabel *tagsList;

@property (nonatomic, strong) NSString *tagsString;

@end

@implementation EditImageTagsTableViewCell

- (void)awakeFromNib
{
    // Initialization code
    [super awakeFromNib];

    self.backgroundColor = [UIColor piwigoColorBackground];
	
	self.tagsString = @"";
	self.tagsLabel.text = NSLocalizedString(@"editImageDetails_tags", @"Tags");
    self.tagsLabel.font = [UIFont piwigoFontNormal];
    self.tagsList.font = [UIFont piwigoFontNormal];
}

-(void)setTagsString:(NSString *)tagsString
{
	_tagsString = tagsString;

	if (tagsString.length <= 0) {
		self.tagsList.text = NSLocalizedString(@"none", @"none");
	}
	else {
		self.tagsList.text = tagsString;
	}
}

-(void)setTagList:(NSArray*)tags inColor:(UIColor *)color
{
	self.tagsString = [[TagsData sharedInstance] getTagsStringFromList:tags];
    self.tagsLabel.textColor = [UIColor piwigoColorRightLabel];
    self.tagsList.textColor = color;
}

@end

File: piwigo/Image/Parameters/Cells/EditImageTagsTableViewCell.swift
//
//  EditImageTagsTableViewCell.swift
//  piwigo
//
//  Created by Spencer Baker on 2/18/15.
//  Copyright (c) 2015 bakercrew. All rights reserved.
//
//  Converted to Swift 5.3 by Eddy Lelièvre-Berna on 29/08/2021.
//

import UIKit
import piwigoKit

class EditImageTagsTableViewCell: UITableViewCell {
    
    @IBOutlet private weak var tagsLabel: UILabel!
    @IBOutlet private weak var tagsList: UILabel!

    private var _tagsString: String?
    var tagsString: String? {
        get {
            return _tagsString
        }
        set(tags) {
            _tagsString = tags

            if (tagsString?.count ?? 0) <= 0 {
                tagsList.text = NSLocalizedString("none", comment: "none")
            } else {
                tagsList.text = tagsString
            }
        }
    }

    override func awakeFromNib() {
        // Initialization code
        super.awakeFromNib()

        // Cell background
        backgroundColor = .piwigoColorBackground()

        // Cell label
        tagsLabel.text = NSLocalizedString("editImageDetails_tags", comment: "Tags")

        // Cell tags list
        tagsString = ""
    }

    func config(withList tags: Set<Tag>, inColor color: UIColor?) {
        // Set colours
        tagsLabel.textColor = UIColor.piwigoColorLeftLabel()
        tagsList.textColor = color

        // Compile list of tags
        tagsString = String(tags.compactMap({"\($0.tagName), "}).reduce("", +).dropLast(2))
    }
}
